### PR TITLE
Transaction process 1/4: Add new schema, save new data to new schema

### PR DESCRIPTION
**Pull request by rap1ds**
- [x] Create a TransactionProcess model
- [x] Each TransactionType has one TransactionProcess. Actually, the relation is `belongs_to` since the foreign key is in the TransactionType model
- [x] Save TransactionProcess in TransactionTypeCreator

### CODE REVIEW

File: db/migrate/20150303134630_create_transaction_processes.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class CreateTransactionProcesses < ActiveRecord::Migration
+  def change
+    create_table :transaction_processes do |t|
+      t.string :process, null: false
```
Comment (Gnito) [inline db/migrate/20150303134630_create_transaction_processes.rb]: btw. what do you think, should this string be limited from default 255?

**Comment by Gnito** [comment]
:+1:

File: test/helper_modules.rb [hunk 2]
```diff
@@ -48,11 +63,13 @@ def self.load_test_categories_and_transaction_types_to_db(community)
 
     def self.load_categories_and_transaction_types_to_db(community, transaction_types, categories)
       # Load transaction types
-      transaction_types.each do |type, translations|
+      transaction_types.each do |type, opts|
 
         transaction_type = Object.const_get(type.to_s).create!(:type => type, :community_id => community.id)
+        transaction_type.create_transaction_process(process: opts[:process])
```
Comment (sktoiva) [inline test/helper_modules.rb]: It'd be better to use ```create_transaction_process!``` as it throws instead of returning false on invalid association object.

File: test/helper_modules.rb [hunk 3]
```diff
@@ -48,11 +63,13 @@ def self.load_test_categories_and_transaction_types_to_db(community)
...
+        transaction_type.save!
```
Comment (sktoiva) [inline test/helper_modules.rb]: Why is this needed?

File: app/services/transaction_type_creator.rb [hunk 4]
```diff
@@ -110,4 +112,15 @@ def use_in_category(category, transaction_type)
...
 
+  # This method is only temporary implementation for the transaction_process change
+  def create_transaction_process!(community, transaction_type, payment_gateway_available)
+    payment_gateway_available = payment_gateway_available || MarketplaceService::Community::Query.payment_type(community.id).present?
+    process = ListingsController.select_payment_process(
+          price_field: transaction_type.price_field?,
+          preauthorize: transaction_type.preauthorize_payment?,
+          payment_gateway_available: payment_gateway_available)
+
+    transaction_type.create_transaction_process({ process: process })
```
Comment (sktoiva) [inline app/services/transaction_type_creator.rb]: Better to use ```create_transaction_process!``` as it throws.

File: test/helper_modules.rb (hunk 3 above)
Comment (rap1ds) [inline test/helper_modules.rb]: You mean `save!`?

The relation between TransactionType and TransactionProcess is so that TransactionType _belongs\_to_ TransactionProcess, which means that the foreign key is store to TransactionType.transaction\_process\_id

When `transaction_type.create_transaction_process` is called, a new TransactionProcess is instantiated AND saved to database. The id of the newly created TransactionProcess is set to TransactionType but NOT saved. That's why `save` is needed.

This wouldn't be the case if the relationship was TransactionType _has\_one_ TransactionProcess. The calling `transaction_type.create_transaction_process` would instantiate and save the TransactionProcess, including the foreign key `transaction_type_id`
